### ISSUE TITLE
[Popover] Content width CSS Variable is not responsive

### ISSUE THREAD

**Issue by ng-hai** [radix-ui/primitives#2019]
### Current Behavior

According to the docs, I used `--radix-popover-content-available-width` to apply for Content `max-width`, then this happened after scaling the viewport.

https://user-images.githubusercontent.com/19520278/224609336-54ad2d8f-d1df-4915-adbe-92f45dbd9262.mov

### Expected behavior

`--radix-popover-content-available-width` should be re-calculated when changing viewport width, like `--radix-popover-content-available-height` behavior.

### Reproducible example

I cannot create a full example on Codesandbox. This is how I do styling with Tailwind, this code also use [custom APIs](https://www.radix-ui.com/docs/primitives/components/popover#custom-apis) as the docs implementation.

<img width="1264" alt="image" src="https://user-images.githubusercontent.com/19520278/224610527-a0590ff0-8b77-4040-b8dc-8207f8bf687a.png">

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Radix Package(s) | @radix-ui/react-popover|1.0.5|
| React            |      |18.2|
| Browser          |Edge|110.0.1587.69 (Official build) (arm64)|
| Assistive tech   |         |         |
| Node             |     |16.17|
| npm/yarn         | yarn|         |
| Operating System | macOS|13.2.1|

**Comment by h2thien** [comment]
Hi @ng-hai for starting, can you check in your dev tool whether `--radix-popper-available-width` is updated correctly, both while you are shrinking and expanding the viewport?

Unfortunately GitHub doesn't allow video bigger than 10MB so I sent you a recording of the case via mail

**Comment by ng-hai** [comment]
I could confirm that `--radix-popper-available-width` did not change.

<img width="1840" alt="image" src="https://user-images.githubusercontent.com/19520278/224622699-31c2390b-548c-443b-8a60-b1c78b2f5eb1.png">

After shrinking and expanding
<img width="1840" alt="image" src="https://user-images.githubusercontent.com/19520278/224622854-b3c94e38-4579-45c1-af29-f54c3813cc83.png">

**Comment by h2thien** [comment]
Does this happen only in the latest version? Any chance https://github.com/radix-ui/primitives/pull/1942 is related?

**Comment by ng-hai** [comment]
Yes, I'm using the latest version. Before that PR, there is no `--radix-popover-content-available-width` variable.

**Comment by benoitgrelard** [comment]
As far as I can tell it does update:

https://user-images.githubusercontent.com/1539897/224700388-b61e3022-19fc-4b8e-9f13-12d0500b9002.mp4

**Comment by ng-hai** [comment]
Finally have time to reproduce it
https://codesandbox.io/s/boring-bhaskara-ei9lnd?file=/src/App.tsx

**Comment by benoitgrelard** [comment]
Would you be able to create a simpler example of your issue, this is rather convoluted with a lot of things getting in the way…

**Comment by atomiks** [comment]
I took a look at this and it's a bug in Floating UI for center aligned placements which is still present in latest, seems hard to fix as the measurements in this case depend on the size of the floating element itself, which when it resizes, causes the issue :\

Fixes I'm aware of:

- Most practical right now: avoid the variable and use regular CSS: `max-width: calc(100% - 2 * VIEWPORT_PADDING_PX)`
- If Radix upgraded to Floating UI v1: `align="end"` works in this case, but doesn't seem to work with v0 of the library from what I tested
- Radix could add a middleware to the beginning of the array that removes the size CSS variables before every update, making sure that it only runs once per lifecycle so the popover has its natural size to be measured ([demo](https://codesandbox.io/s/recursing-hypatia-f3f3bq?file=/src/App.tsx))
- Someone could help fix the issue in the library so we don't need any hacks or workarounds 😄 

```js
{
  name: "removeVariables",
  fn({ elements, middlewareData }) {
    // skip this running again after the first lifecycle run
    if (middlewareData.removeVariables) return {};

    elements.floating.style.removeProperty(...);

    // re-measure elements and restart lifecycle
    return {
      reset: {
        rects: true
      }
    };
  }
},
```

**Comment by benoitgrelard** [comment]
Fixed when we updated to latest floating-ui #2032
